### PR TITLE
Add Trove classifier for Python 3.13

### PR DESCRIPTION
**Pull request by hugovk**
```toml
dependencies = [
  "cffi>=v1.17.0rc1", # remove once v1.17.0 is out; add 3.13 classifier above (see #127)
```

cffi 1.17.1 is now out: https://pypi.org/project/cffi/ 

Also replace Flake8 with Ruff, and add some extra linting.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -50,5 +47,26 @@ tag-pattern = '^cherry-picker-(?P<version>[vV]?\d+(?:\.\d+){0,2}[^\+]*)(?:\+.*)?
 [tool.hatch.version.raw-options]
 local_scheme = "no-local-version"
 
-[tool.isort]
-profile = "black"
+[tool.ruff]
+fix = true
+
+lint.select = [
```
Comment (AlexWaygood) [inline pyproject.toml:53]: did you want to enable `isort` rules here too? I saw that you removed `PCQA/isort` from pre-commit

File: pyproject.toml [hunk 2]
```diff
@@ -50,5 +47,26 @@ tag-pattern = '^cherry-picker-(?P<version>[vV]?\d+(?:\.\d+){0,2}[^\+]*)(?:\+.*)?
...
+]
+
+lint.ignore = [
+  "S101", # Use of assert detected
+  "S404", # subprocess module is possibly insecure
+  "S603", # subprocess call: check for execution of untrusted input
+]
+lint.per-file-ignores."cherry_picker/test_*.py" = [
+  "S101", # Asserts allowed in tests
+]
```
Comment (AlexWaygood) [inline pyproject.toml:69]: hmm, is this section necessary? It looks like you already have `S101` in `lint.ignore`

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,60 +1,42 @@
...
-    rev: 23.12.1
+    rev: 24.8.0
     hooks:
       - id: black
 
-  - repo: https://github.com/PyCQA/isort
-    rev: 5.13.2
-    hooks:
-      - id: isort
-        args: [--add-import=from __future__ import annotations]
```
Comment (AlexWaygood) [inline .pre-commit-config.yaml:17]: You can also do this with Ruff (if you like) via the `tool.ruff.lint.isort.required-imports` setting: https://docs.astral.sh/ruff/settings/#lint_isort_required-imports

Comment (hugovk) [reply .pre-commit-config.yaml:17]: Re-added!

File: pyproject.toml (hunk 2 above)
Comment (AlexWaygood) [reply pyproject.toml:77]:
```suggestion
```

**Review by AlexWaygood — APPROVED** [review]

**Pull request merged**